src/openforms/config/models.py

```diff
@@ -661,19 +661,35 @@ def as_dict(self):
 class CSPSettingManager(models.Manager.from_queryset(CSPSettingQuerySet)):
     @transaction.atomic
     def set_for(
-        self, obj: models.Model, settings: list[tuple[CSPDirective, str]]
+        self,
+        obj: models.Model,
+        settings: list[tuple[CSPDirective, str]],
+        identifier: str | None = None,
```

Comment:
```suggestion
        identifier: str = "",
```
seems to make more sense to me so that you have a consistent type

Reply:
True, this is a leftover from the previous solution where I wanted to have the ability to _not_ filter against `identifier` (by leaving it as `None`). This now doesn't make sense to have it.

Reply:
My bad this is still relevant:

```python
predicate = {
    "content_type": get_content_type_for_model(obj),
    "object_id": str(obj.id),
}
if identifier is not None:
    predicate["identifier"] = identifier
```

The idea was that if you don't specify an `identifier`, all `CSPSettings` (matching the content object) would be deleted. You'd have to explicitly specify `identifier=""` if you only want to delete the ones with an empty/unset identifier (but not the ones _with_ an identifier!)

Is this desirable?

Reply:
in practice that behaviour is not used, right? if we don't have a need for identifier, records will never be set/created with an identifier, so filtering on empty string will effectively delete them. for analytics we need it, and we don't call that one without identifier provided.